Python/Product/Analysis/AnalysisLogWriter.cs

```diff
@@ -222,36 +223,12 @@ public struct LogItem {
             }
         }
 
-        private FileStream OpenRawLogFile(bool appendOnly, bool create) {
-            // Retry for up to one second
-            for (int retries = 100; retries > 0; --retries) {
-                try {
-                    return appendOnly ? new FileStream(_outputFile, FileMode.Append, FileAccess.Write, FileShare.Read)
-                                      : new FileStream(_outputFile, create ? FileMode.OpenOrCreate : FileMode.Open, FileAccess.ReadWrite, FileShare.None);
-                } catch (FileNotFoundException) when (!create) {
-                    return null;
-                } catch (DirectoryNotFoundException) when (!create) {
-                    return null;
-                } catch (IOException) {
-                    var dir = Path.GetDirectoryName(_outputFile);
-                    try {
-                        Directory.CreateDirectory(dir);
-                    } catch (IOException) {
-                        // Cannot create directory for DB, so just bail out
-                        return null;
-                    }
-                    Thread.Sleep(10);
-                }
-            }
-            return null;
-        }
-
         private TextWriter OpenLogFile() {
             if (string.IsNullOrEmpty(_outputFile)) {
                 return null;
             }
             var enc = new UTF8Encoding(false);
-            var stream = OpenRawLogFile(true, true);
+            var stream = PathUtils.OpenWithRetry(_outputFile, FileMode.Append, FileAccess.Write, FileShare.Read);
```

Comment:
Why are we trying to get file access several times?

Reply:
Because we've seen random IO failures in the past that work on retry (presumably AV or disk tools, but it's near impossible to diagnose).